### PR TITLE
Paid subscriber import: enable for Jetpack sites, hide content step

### PR DESCRIPTION
**Pull request by simison**
Depends on backend change: D167682-code

## Proposed Changes

* Enables paid subscriber importer for Jetpack sites

Differences from WP.com version:
- Hides "Content" step; the content importer on Jetpack site works significantly differently [via .org plugin](https://wordpress.org/plugins/substack-importer/) so for Jetpack sites we need to focus on free and paid subscribers.
- No "View content" final CTA
- Point to Jetpack docs instead of WP.com docs

Not in PR though good follow ups:
- See if we need step counter at all
- Hide "skip" since there's no point skipping a single step
- URL initially has `content` bit instead of `subscribers`

### Before

Screenshot 2024-12-03 at 12 03 32

image

### After

image
Screenshot 2024-12-03 at 13 19 04
Screenshot 2024-12-03 at 13 19 19
Screenshot 2024-12-03 at 13 19 40
Screenshot 2024-12-03 at 13 20 47
Screenshot 2024-12-03 at 14 28 34

## Why are these changes being made?

*

## Testing Instructions

To test the modal:

* Go https://cloud.jetpack.com/subscribers but localhost or [Live version](https://github.com/Automattic/wp-calypso/pull/97005#issuecomment-2514327887)
* Click "Add subscribers"; you should see Substack option

To test the importer in Calypso:

* Apply D167682-code and sandbox public api
* Open `/import/newsletter/substack/JETPACK_SITE_SLUG` in localhost Calypso or [Live version](https://github.com/Automattic/wp-calypso/pull/97005#issuecomment-2514327887)
* Go through importing substack subscribers process

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/my-sites/importer/newsletter/subscribers/paid-subscribers/connect-stripe.tsx [hunk 1]
```diff
@@ -64,7 +64,7 @@ export default function ConnectStripe( {
 					onClick={ () => {
 						recordTracksEvent( 'calypso_paid_importer_connect_stripe' );
 					} }
-					ariaLabel={ __( 'Connect Stripe' ) }
+					aria-label={ __( 'Connect Stripe' ) }
```
Comment (simison) [inline client/my-sites/importer/newsletter/subscribers/paid-subscribers/connect-stripe.tsx:67]: Fixes warning:

Screenshot 2024-12-03 at 13 39 51

File: client/my-sites/importer/newsletter/subscribers/paid-subscribers/connect-stripe.tsx [hunk 2]
```diff
@@ -50,7 +50,7 @@ export default function ConnectStripe( {
 			<p>
 				{ createInterpolateElement(
 					__(
-						"To migrate your <strong>paid subscribers</strong> to WordPress.com, make sure you're connecting the <strong>same</strong> Stripe account you use with Substack."
```
Comment (simison) [inline client/my-sites/importer/newsletter/subscribers/paid-subscribers/connect-stripe.tsx:53]: @crisbusquets I could remove the WordPress.com bit just for Jetpack sites, or all sites for simplicity. WDYT? We could also leave it in of course but felt a bit odd in Jetpack site context.

**Review by lezama — APPROVED** [review]
Looking good!

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/17045801

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @simison for including a screenshot in the description! This is really helpful for our translators.

**Pull request merged**